aggregator/utils/utils.py

```diff
@@ -262,37 +300,34 @@ async def query_service(service, params, access_token, ws=None):
     """Query service with params."""
     LOG.debug("Querying service.")
     headers = {}
-
     if access_token:
         headers.update({"Authorization": f"Bearer {access_token}"})
-
+    endpoint = await find_query_endpoint(service, params)
     # Pre-process query string into payload format
-    data = await pre_process_payload(service[1], params)
-
-    # Query service in a session
-    async with aiohttp.ClientSession() as session:
-        try:
-            async with session.post(service[0], json=data, headers=headers, ssl=await request_security()) as response:
-                LOG.info(f"POST query to service: {service[0]}")
-                # On successful response, forward response
-                if response.status == 200:
-                    return await _service_response(response, ws)
-
-                elif response.status == 405:
-                    return await _get_request(session, service, params, headers, ws)
-
-                else:
-                    # HTTP errors
-                    error = {"service": service[0], "queryParams": params, "responseStatus": response.status, "exists": None}
-                    LOG.error(f"Query to {service} failed: {response}.")
-                    if ws is not None:
-                        return await ws.send_str(json.dumps(error))
+    if endpoint is not None:
+        data = await pre_process_payload(endpoint[1], params)
+
+        # Query service in a session
+        async with aiohttp.ClientSession() as session:
+            try:
+                async with session.post(endpoint[0], json=data, headers=headers, ssl=await request_security()) as response:
+                    LOG.info(f"POST query to service: {endpoint}")
+                    # On successful response, forward response
+                    if response.status == 200:
+                        return await _service_response(response, ws)
+                    elif response.status == 405:
+                        return await _get_request(session, endpoint, params, headers, ws)
                     else:
-                        return error
-
-        except Exception as e:
-            LOG.debug(f"Query error {e}.")
-            web.HTTPInternalServerError(text="An error occurred while attempting to query services.")
+                        # HTTP errors
+                        error = {"service": endpoint, "queryParams": params, "responseStatus": response.status, "exists": None}

```

Comment:
```suggestion
                        error = {"service": endpoint[0], "queryParams": params, "responseStatus": response.status, "exists": None}

```

should fix integration tests